### PR TITLE
Splits API interface to API and VibeRestAPI

### PR DESCRIPTION
**Pull request by denizzzka**
It is only cosmetic change what isn't touches anything but makes things easier to understand and maybe more flexible for further wire protocols implementation.

### CODE REVIEW

File: source/agora/network/RestApi.d [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        Params:
+            block_height = Height of the block that contains the transaction hash
+            hash         = transaction hash
+
+        Returns:
+            the array of hashes which form the merkle path
+
+    ***************************************************************************/
+
+    public Hash[] getMerklePath (ulong block_height, Hash hash);
```
Comment (Geod24) [inline source/agora/network/RestApi.d]: Those functions don't need to be redefined since they are in the base API.

**Comment by Geod24** [comment]
This makes much more sense, yes. However, I don't see it bringing any value at the moment. The REST attributes being on the `interface` can be seen as inelegant, but they are not so annoying that they require so much duplication.

**Comment by Geod24** [comment]
To clarify: If it becomes a necessity in the future, we can do this. But there's no clear use case for it at the moment.

File: source/agora/network/RestApi.d (hunk 1 above)
Comment (denizzzka) [reply source/agora/network/RestApi.d]: Fixed

File: source/agora/network/RestApi.d [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        Returns:
+            Return true if the node has this transaction hash.
+
+        API:
+            GET /hasTransactionHash
+
+    ***************************************************************************/
+
+    @method(HTTPMethod.GET)
+    public bool hasTransactionHash (Hash tx);
```
Comment (Geod24) [inline source/agora/network/RestApi.d]: Does `override` works here ?

Comment (denizzzka) [reply source/agora/network/RestApi.d]: Yes, works with and without it

**Comment by denizzzka** [comment]
Weird PR, yes. But let it be - for some elusive reason I’m so calmer with it. :-)

**Comment by Geod24** [comment]
> To clarify: If it becomes a necessity in the future, we can do this. But there's no clear use case for it at the moment.

**Comment by denizzzka** [comment]
Value of this PR is in adding clarity.

**Comment by AndrejMitrovic** [comment]
It's not adding any clarity whatsoever. There should be a single interface defining all the public-facing API entry-points for the node.

**Comment by denizzzka** [comment]
> It's not adding any clarity whatsoever. There should be a single interface defining all the public-facing API entry-points for the node.

...and another one for Vibe rest generator to distinguish between API and Vibe.d REST API implementation (what accidentally looks now like D's interface description block equal to API interface block).

This PR leads to avoid exposing symbols that are unnecessary in a particular interface, add necessary ones, change their ​​if necessary (https://github.com/bpfkorea/agora/pull/150#discussion_r311352084). Say, different wiring schemes can use different addresses or may have additional calls.

Related: https://github.com/Geod24/localrest/pull/20 - in my opinion, including therefore no one knew that LocalRest actually is not implements REST

**Comment by Geod24** [comment]
As there is not any consensus in favor of this, I am going to close this.
As mentioned in our discussions, we can re-discuss the issue once the TCP communication protocol is ready.

**Pull request closed without merging**